src/helpers/formattedStrings.ts

```diff
@@ -58,19 +58,35 @@ export const removeNonDigits = (value: string): string => {
 }
 
 export const snakeCaseToCamelCase = (s: string) => {
-  return s.replace(/_([a-z])/g, (_: string, char: string) => char.toUpperCase())
+  // First, identify and preserve two-letter state codes
+  const parts = s.split('.')
+  const processedParts = parts.map((part, index) => {
+    // Check if this part is a two-letter state code (assuming it's in the second position)
+    if (index === 1 && part.length === 2) {
+      return part.toUpperCase() // Convert to uppercase
+    }
+    // For other parts, convert to camelCase
+    return part.replace(/_([a-z])/g, (_: string, char: string) => char.toUpperCase())
+  })
```

Comment:
I'm concerned here with tax specific logic being placed inside of this general utility function for snake to camel case conversion.

Have we considered breaking the state capitalization functionality out from this and placing it in either a tax specific string utility that uses the snake to camel conversion internally instead of baking the state capitalization logic into this function?

Reply:
So yes, I've considered it. The issue is we receive this data back from the API in snake case and need to get it into both a valid format for speakEasy while respecting our API rules. 

I agree there is smell here, but i'm not quite sure there is a more elegant way to solve this at the moment. The best solution IMO would be to stay snake_case but I think Speakeasy took that choice from us.